tw-context/src/main/java/com/transferwise/common/context/TwContext.java

```diff
@@ -35,9 +36,14 @@ public class TwContext {
   private static final ThreadLocal<Optional<TwContext>> contextTl = new ThreadLocal<>();
   private static final List<TwContextExecutionInterceptor> interceptors = new CopyOnWriteArrayList<>();
   private static final List<TwContextAttributePutListener> attributePutListeners = new CopyOnWriteArrayList<>();
-  private static final TwContext ROOT_CONTEXT = new TwContext(null, true);
   private static final RateLimiter throwableLoggingRateLimiter = RateLimiter.create(2);
 
+  static final TwContext ROOT_CONTEXT = new TwContext(null, true);
+
+  static {
```

Comment:
Probably might be better to put this logic into the starter package and using an autoconfiguration mechanism for that. Using the static block is probably too hard coded in this case.